execution-engine/src/hcall/wasmi.rs

```diff
@@ -562,59 +546,58 @@ impl WasmiHostProvisioningState {
         let size: u32 = args.nth(2);
 
         match self.get_memory() {
-            None => Err(FatalHostError::NoMemoryRegistered),
-            Some(memory) => match self.get_current_data_source(index as usize) {
-                None => return Ok(VeracruzError::BadInput),
-                Some(frame) => {
-                    let data = frame.get_data();
-
-                    if data.len() > size as usize {
-                        return Ok(VeracruzError::DataSourceSize);
-                    }
+            None => Err(FatalEngineError::NoMemoryRegistered),
+            Some(memory) => 
+            {
+                //TODO FILL in appropriate principal
+                let data = self.read_file(&self.program,&format!("input-{}",index))?.ok_or(format!("File input-{} cannot be found",index))?;
+                if data.len() > size as usize {
+                    return Ok(EngineReturnCode::DataSourceSize);
+                }
+                if let Err(_) = memory.set(address, &data) {
```

Comment:
Same comment as above about preferring `.is_err()` over the dummy pattern matching.